crates/interfaces/src/p2p/headers/downloader.rs

```diff
@@ -87,16 +89,3 @@ pub fn validate_header_download(
         .map_err(|error| DownloadError::HeaderValidation { hash: parent.hash(), error })?;
     Ok(())
 }
-
-/// Ensures that the given `parent` header is the actual parent of the `header`
-pub fn ensure_parent(header: &SealedHeader, parent: &SealedHeader) -> DownloadResult<()> {
-    if !(parent.hash() == header.parent_hash && parent.number + 1 == header.number) {
-        return Err(DownloadError::MismatchedHeaders {
-            header_number: header.number,
-            parent_number: parent.number,
-            header_hash: header.hash(),
-            parent_hash: parent.hash(),
-        })
-    }
-    Ok(())
-}
```

Comment:
`validate_header_against_parent` already validates the congruency of block numbers. don't see any reason why it shouldn't validate parent hash as well

Reply:
Do we also want to alter the comment here? https://github.com/paradigmxyz/reth/blob/529c8e003c461873d6c90d14343bc2f087495b1c/crates/interfaces/src/consensus.rs#L19C1-L32